applicationset/services/plugin/utils.go

```diff
@@ -0,0 +1,37 @@
+package plugin
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/argoproj/argo-cd/v2/common"
+	log "github.com/sirupsen/logrus"
+)
+
+// ReplaceStringSecret checks if given string is a secret key reference ( starts with $ ) and returns corresponding value from provided map
+func ReplaceStringSecret(val string, secretValues map[string]string) string {
```

Comment:
I feel like we probably already have a utility function elsewhere to implement this.

Reply:
Yes indeed, it is here :

/argo-cd/util/settings/settings.go

I didn't use much of the project's root internal utils, and since it wasn't part of applicationset, I didn't know its impact, or if it would have been modified, so I duplicated it.

If you think we can use it, we do.

Reply:
I'd try using the same code. If it ends up being gnarly, duplication will be fine.

Reply:
Will try to do this in my PR